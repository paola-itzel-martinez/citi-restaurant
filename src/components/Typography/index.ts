export { default as Typography } from './Typography';
export * as TypographyStyles from './Typography.styles';
