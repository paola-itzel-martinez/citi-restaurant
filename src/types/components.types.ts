export enum ButtonVariant {
  CARD = 'CARD',
  MENU = 'MENU',
}

export enum DividerVariant {
  HORIZONTAL = 'HORIZONTAL',
  VERTICAL = 'VERTICAL',
}

export enum Portals {
  MODAL = 'MODAL',
}

export enum TypographyVariant {
  BUTTON = 'BUTTON',
  BUTTON_MENU = 'BUTTON_MENU',
  NORMAL = 'NORMAL',
  SMALL = 'SMALL',
  TITLE = 'TITLE',
}

export enum TypographyFamily {
  GENERAL = "'Roboto',sans-serif",
}

export type CSSRule = number | string;

export type CSSMarginRules = {
  m?: CSSRule;
  mx?: CSSRule;
  my?: CSSRule;
  mt?: CSSRule;
  mr?: CSSRule;
  mb?: CSSRule;
  ml?: CSSRule;
};

export type CSSPaddingRules = {
  p?: CSSRule;
  px?: CSSRule;
  py?: CSSRule;
  pt?: CSSRule;
  pr?: CSSRule;
  pb?: CSSRule;
  pl?: CSSRule;
};

export type CSSSpacingRules = {
  width?: CSSRule;
  minWidth?: CSSRule;
  maxWidth?: CSSRule;
  height?: CSSRule;
  minHeight?: CSSRule;
  maxHeight?: CSSRule;
};
