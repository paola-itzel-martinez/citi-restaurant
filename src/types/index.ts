export * from './components.types';
export * from './mediaQueries.types';
export * from './yelpAPI.types';
