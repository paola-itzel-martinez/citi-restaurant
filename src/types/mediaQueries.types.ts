export enum SCREEN_SIZE {
  DESKTOP = 'DESKTOP',
  TABLET = 'TABLET',
  MOBILE = 'MOBILE',
}
