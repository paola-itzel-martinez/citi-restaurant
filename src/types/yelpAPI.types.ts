export type Category = {
  title: string;
  alias: string;
};

export type Business = {
  id: string;
  categories: Category[];
  image_url: string;
  name: string;
  rating: number;
  url: string;
  price?: string;
};
