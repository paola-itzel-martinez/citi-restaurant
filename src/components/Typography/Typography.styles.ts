import styled from 'styled-components';
import theme from 'styled-theming';
import { MEDIA_MOBILE, styledCSSMarginRules } from '../../styled';
import { TypographyFamily, TypographyVariant } from '../../types';
import { TypographyCSS } from './Typography.types';

const fontSizeDesktop = theme.variants('mode', 'variant', {
  [TypographyVariant.BUTTON]: { default: '16px' },
  [TypographyVariant.BUTTON_MENU]: { default: '22px' },
  [TypographyVariant.NORMAL]: { default: '16px' },
  [TypographyVariant.SMALL]: { default: '14px' },
  [TypographyVariant.TITLE]: { default: '24px' },
});

const fontSizeMobile = theme.variants('mode', 'variant', {
  [TypographyVariant.BUTTON]: { default: '14px' },
  [TypographyVariant.BUTTON_MENU]: { default: '18px' },
  [TypographyVariant.NORMAL]: { default: '20px' },
  [TypographyVariant.SMALL]: { default: '14px' },
  [TypographyVariant.TITLE]: { default: '20px' },
});

const fontWeight = theme.variants('mode', 'variant', {
  [TypographyVariant.BUTTON]: { default: '400' },
  [TypographyVariant.BUTTON_MENU]: { default: '400' },
  [TypographyVariant.NORMAL]: { default: '400' },
  [TypographyVariant.SMALL]: { default: '400' },
  [TypographyVariant.TITLE]: { default: '700' },
});

export const Wrapper = styled.p<TypographyCSS>`
  color: ${({ color }) => color};
  font-family: ${TypographyFamily.GENERAL};
  font-weight: ${fontWeight};
  letter-spacing: normal;
  text-align: ${({ textAlign }) => (textAlign ? textAlign : undefined)};
  font-size: ${fontSizeDesktop};
  text-overflow: ${({ hasEllipsis }) => hasEllipsis && 'ellipsis'};
  overflow: ${({ hasEllipsis }) => hasEllipsis && 'hidden'};
  white-space: ${({ hasEllipsis }) => hasEllipsis && 'nowrap'};
  padding: 0;
  margin: 0;
  min-width: 20px;

  ${(margin) => styledCSSMarginRules(margin)};

  @media ${MEDIA_MOBILE} {
    font-size: ${fontSizeMobile};
  }
`;
