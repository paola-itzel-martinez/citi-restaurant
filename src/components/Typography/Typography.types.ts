import { CSSMarginRules, TypographyVariant } from '../../types';

export type GetAsType = (params: {
  variant: TypographyVariant;
}) => keyof JSX.IntrinsicElements;

export interface TypographyCSS extends CSSMarginRules {
  color?: string;
  hasEllipsis?: boolean;
  textAlign?: 'center' | 'start';
  variant?: TypographyVariant;
}

export interface TypographyProps extends TypographyCSS {
  children: any;
}
