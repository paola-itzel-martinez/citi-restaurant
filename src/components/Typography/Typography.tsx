import React from 'react';
import { palette } from '../../styled';
import { TypographyVariant } from '../../types';
import { Wrapper } from './Typography.styles';
import { GetAsType, TypographyProps } from './Typography.types';

const getAsType: GetAsType = ({ variant }) => {
  if (variant === TypographyVariant.TITLE) return 'h1';

  return 'p';
};

const Typography = ({
  children,
  textAlign,
  color = palette.NORMAL_TEXT,
  hasEllipsis = false,
  variant = TypographyVariant.NORMAL,
}: TypographyProps) => {
  const as = getAsType({ variant });

  return (
    <Wrapper
      as={as}
      variant={variant}
      color={color}
      hasEllipsis={hasEllipsis}
      textAlign={textAlign}>
      {children}
    </Wrapper>
  );
};

export default React.memo(Typography);
