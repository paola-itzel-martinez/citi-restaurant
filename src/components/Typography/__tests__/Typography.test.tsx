import React from "react";
import { render, screen } from "../../../../tests";
import { TypographyVariant } from "../../../types";
import { TypographyProps } from "../Typography.types";
import { Typography } from "..";

const CHILDREN = "text";
const GeneralComponent = (props: Omit<TypographyProps, "children">) => (
  <Typography {...props}>{CHILDREN}</Typography>
);

describe("<StarsRating />", () => {
  it(`Should have a specific css rules if the variant is ${TypographyVariant.BUTTON}`, () => {
    const { container } = render(
      <GeneralComponent variant={TypographyVariant.BUTTON} />
    );

    expect(container).toMatchSnapshot();
  });

  it(`Should have a specific css rules if the variant is ${TypographyVariant.BUTTON_MENU}`, () => {
    const { container } = render(
      <GeneralComponent variant={TypographyVariant.BUTTON_MENU} />
    );

    expect(container).toMatchSnapshot();
  });

  it(`Should have a specific css rules if the variant is ${TypographyVariant.NORMAL}`, () => {
    const { container } = render(
      <GeneralComponent variant={TypographyVariant.BUTTON} />
    );

    expect(container).toMatchSnapshot();
  });

  it(`Should have a specific css rules if the variant is ${TypographyVariant.SMALL}`, () => {
    const { container } = render(
      <GeneralComponent variant={TypographyVariant.BUTTON_MENU} />
    );

    expect(container).toMatchSnapshot();
  });

  it(`Should have a specific css rules if the variant is ${TypographyVariant.TITLE}`, () => {
    const { container } = render(
      <GeneralComponent variant={TypographyVariant.BUTTON_MENU} />
    );

    expect(container).toMatchSnapshot();
  });

  it("Should have a specific css rules if textAlign is assigned", () => {
    render(<GeneralComponent textAlign="center" />);

    expect(screen.getByText(CHILDREN)).toHaveStyleRule("text-align", "center");
  });

  it("Should have a specific css rules if color is assigned", () => {
    render(<GeneralComponent color="orange" />);

    expect(screen.getByText(CHILDREN)).toHaveStyleRule("color", "orange");
  });

  it("Should have a specific css rules if hasEllipsis is truthy", () => {
    render(<GeneralComponent hasEllipsis />);

    expect(screen.getByText(CHILDREN)).toHaveStyle(`
        text-overflow: ellipsis;
        overflow: hidden;
        white-space: nowrap;
    `);
  });

  it("Should have a specific css rules if hasEllipsis is falsy", () => {
    render(<GeneralComponent />);

    expect(screen.queryByText(CHILDREN)).not.toHaveStyleRule(
      "text-overflow",
      "ellipsis"
    );

    expect(screen.queryByText(CHILDREN)).not.toHaveStyleRule(
      "overflow",
      "hidden"
    );

    expect(screen.queryByText(CHILDREN)).not.toHaveStyleRule(
      "white-space",
      "nowrap"
    );
  });
});
