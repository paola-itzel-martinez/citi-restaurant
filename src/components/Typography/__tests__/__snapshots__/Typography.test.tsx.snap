// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StarsRating /> Should have a specific css rules if the variant is BUTTON 1`] = `
.c0 {
  color: #415058;
  font-family: 'Roboto',sans-serif;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  padding: 0;
  margin: 0;
  min-width: 20px;
}

@media (min-width:0px) and (max-width:599px) {

}

<div>
  <p
    class="c0"
    color="#415058"
  >
    text
  </p>
</div>
`;

exports[`<StarsRating /> Should have a specific css rules if the variant is BUTTON_MENU 1`] = `
.c0 {
  color: #415058;
  font-family: 'Roboto',sans-serif;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  padding: 0;
  margin: 0;
  min-width: 20px;
}

@media (min-width:0px) and (max-width:599px) {

}

<div>
  <p
    class="c0"
    color="#415058"
  >
    text
  </p>
</div>
`;

exports[`<StarsRating /> Should have a specific css rules if the variant is NORMAL 1`] = `
.c0 {
  color: #415058;
  font-family: 'Roboto',sans-serif;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  padding: 0;
  margin: 0;
  min-width: 20px;
}

@media (min-width:0px) and (max-width:599px) {

}

<div>
  <p
    class="c0"
    color="#415058"
  >
    text
  </p>
</div>
`;

exports[`<StarsRating /> Should have a specific css rules if the variant is SMALL 1`] = `
.c0 {
  color: #415058;
  font-family: 'Roboto',sans-serif;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  padding: 0;
  margin: 0;
  min-width: 20px;
}

@media (min-width:0px) and (max-width:599px) {

}

<div>
  <p
    class="c0"
    color="#415058"
  >
    text
  </p>
</div>
`;

exports[`<StarsRating /> Should have a specific css rules if the variant is TITLE 1`] = `
.c0 {
  color: #415058;
  font-family: 'Roboto',sans-serif;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  padding: 0;
  margin: 0;
  min-width: 20px;
}

@media (min-width:0px) and (max-width:599px) {

}

<div>
  <p
    class="c0"
    color="#415058"
  >
    text
  </p>
</div>
`;
